### DISCUSSION TITLE
Can't filter result with businessPhones field

### DISCUSSION THREAD

**Discussion by Niko-F** [YannickRe/spfx-msgraph-peoplesearch#47] [Q&A] [answered]
Hi,

We use people search application to display all phones numbers of the company.
I would like to add a $filter parameter value to display only result when 'businessPhones' filed is not null.
Is it possble to do that ? If yes could you indicate me how ?

I've tried (businessPhones ne null) but it indicate "Invalid filter clause" however the filter (mobilePhone ne null) is ok but is not that we want.

**Comment by YannickRe** [comment, marked as answer]
I'm afraid that will not be possible (or at least I have no idea on how to achieve this). `mobilePhone` is a string property, and that's why `mobilePhone ne null` works, unfortunately the `businessPhone` property is a String collection/array.
This would require a filter that looks more like `businessPhones/any(p:p ne null)`, but this gives the following response:
```
Using 'ne' inside 'any' clause is invalid. Suggestion: consider using the 'not' operator instead. For example: $filter=not groupTypes/any(x:x eq 'Unified')
```
Trying that suggestion with `businessPhones/any(p:p eq null)` returns the following:
```
Using 'eq null' inside 'any' clause is invalid. Example: $filter=proxyAddresses/any(x:x eq null). Consider $filter=proxyAddresses/$count eq 0
```
Trying that suggestion with `businessPhones/$count eq 0` returns:
```
Invalid property used with $count inside $filter
```

I'm out of ideas, I'm sorry ☹. Looks like this is an idea you should post on the Microsoft Graph feedback forum.

**Reply by Niko-F** [reply]
Thanks for your research. 
Not possible with "substringof" or "contains" ?

**Reply by YannickRe** [reply]
As far as I know: no. Sorry.
